milestones/serializers.py

```diff
@@ -0,0 +1,67 @@
+"""
+Data layer serialization operations.  Converts querysets to simple
+python containers (mainly arrays and dicts).
+"""
+from . import models
+
+
+def serialize_milestone(milestone):
+    """
+    Milestone object-to-dict serialization
+    """
+    return {
+        'id': milestone.id,
+        'name': milestone.name,
+        'namespace': milestone.namespace,
+        'description': milestone.description
+    }
+
+
+def serialize_milestone_with_course(course_milestone):
+    """
+    CourseMilestone serialization (composite object)
+    """
+    return {
+        'id': course_milestone.milestone.id,
+        'name': course_milestone.milestone.name,
+        'namespace': course_milestone.milestone.namespace,
+        'description': course_milestone.milestone.description,
+        'course_id': course_milestone.course_id
+    }
+
+
+def serialize_milestone_with_course_content(course_content_milestone):
+    """
+    CourseContentMilestone serialization (composite object)
+    """
+    return {
+        'id': course_content_milestone.milestone.id,
+        'name': course_content_milestone.milestone.name,
+        'namespace': course_content_milestone.milestone.namespace,
+        'description': course_content_milestone.milestone.description,
+        'course_id': course_content_milestone.course_id,
+        'content_id': course_content_milestone.content_id
+    }
+
+
+def serialize_milestones(milestones):
```

Comment:
Creating your own serializers is fine, I think I just preferred the built in serializers since we are dealing with Model Objects.  If you created one, you could simply do something in your data.py such as 
milestone_model = CourseContentMilestone.objects.get(id=milestone.id)
return CourseContentMilestoreSerializer(milestone_model).data
I guess it only comes to mind because we were talking about "user, course, location specific milestones" being a MilestoneRelationshipType between a UserMilestone and a CourseContentMilestone, in which case we may get an advantage to nesting the serializer classes to return one JSON representation of the UserMilestone and its related milestones.